### PR TITLE
Ipmiview: 2.11.0 -> 2.12.0

### PR DESCRIPTION
**Pull request by johnramsden**
###### Motivation for this change

Update version to latest. Old version no longer exists.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

The package does build now after bumping the version, however when using the program and attempting to launch a console with ipmi I get the error 

> Cannot run program “java” error=2, No such file or directory 

I attempted to fix it and looked around but was unable to find the cause of the error. It seems that it's trying to launch the wrong "java". I don't know if this error existed before because I was never able to use the previous version.

### CODE REVIEW

**Comment by Mic92** [comment]
You can use `strace -f -e execve <yourprogram>` in order to find out, which java it is looking for.

**Comment by johnramsden** [comment]
@Mic92 I Just get it failing to run at all with strace.

```
strace -f -e execve IPMIView
execve("/run/current-system/sw/bin/IPMIView", ["IPMIView"], [/* 89 vars */]) = -1 ENOEXEC (Exec format error)
strace: exec: Exec format error
+++ exited with 1 +++
```

File: pkgs/applications/misc/ipmiview/default.nix [hunk 1]
```diff
@@ -4,12 +4,13 @@ assert stdenv.isLinux;
...
    name = "IPMIView-${version}";
-   version = "20151223";
+   version = "2.12.0";
+   buildVersion = "160804";
 
-   src = fetchurl {
-     url = "ftp://ftp.supermicro.com/utility/IPMIView/Linux/IPMIView_V2.11.0_bundleJRE_Linux_x64_${version}.tar.gz";
-     sha256 = "1rv9j0id7i2ipm25n60bpfdm1gj44xg2aj8rnx4s6id3ln90q121";
-   };
+   src = pkgs.fetchurl {
```
Comment (Mic92) [inline pkgs/applications/misc/ipmiview/default.nix]: pkgs is wrong here.

**Comment by Mic92** [comment]
The wrapper was bogus. Please try again.

**Comment by johnramsden** [comment]
@Mic92 No luck, same error.

**Comment by johnramsden** [comment]
OK, wrapper let me run strace. Thanks @Mic92 I'm looking through the output now.

**Comment by johnramsden** [comment]
So it's looking for iKVM.jar, where should I put it? I tried wrapping it but it didn't work.

I tried this:

```
      makeWrapper $out/jre/bin/java $out/bin/iKVM.jar \
        --add-flags "-jar $out/iKVM.jar"

```
and this:

```
      makeWrapper $out/jre/bin/java $out/share/java/iKVM.jar \
        --add-flags "-jar $out/iKVM.jar"
```

```
[pid 32358] execve("/home/john/.local/bin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/home/john/.node_modules/bin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/home/john/bin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/run/wrappers/bin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/run/wrappers/bin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/home/john/.nix-profile/bin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/home/john/.nix-profile/sbin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/home/john/.nix-profile/lib/kde4/libexec/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/nix/var/nix/profiles/default/bin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/nix/var/nix/profiles/default/sbin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/nix/var/nix/profiles/default/lib/kde4/libexec/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/run/current-system/sw/bin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/run/current-system/sw/sbin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] execve("/run/current-system/sw/lib/kde4/libexec/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "Aoy9g1j", "null", "5900", "623", "0", "0"], [/* 91 vars */]) = -1 ENOENT (No such file or directory)
[pid 32358] +++ exited with 255 +++
[pid 32296] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=32358, si_uid=1000, si_status=255, si_utime=0, si_stime=0} ---
```

**Comment by Mic92** [comment]
The problem is not `iKVM.jar`, but you need to extend the `PATH` of IPMIView in the `wrapper` to include `$out/jre/bin/`

File: pkgs/applications/misc/ipmiview/default.nix [hunk 2]
```diff
@@ -24,8 +24,9 @@ stdenv.mkDerivation rec {
    installPhase = ''
      mkdir -p $out/bin
      cp -R . $out/
-     echo "$out/jre/bin/java -jar $out/IPMIView20.jar" > $out/bin/IPMIView
-     chmod +x $out/bin/IPMIView
+     makeWrapper $out/jre/bin/java $out/bin/IPMIView \
+       --prefix PATH : "$out/jre/bin" \
```
Comment (Mic92) [inline pkgs/applications/misc/ipmiview/default.nix]: Try again, with this in place.

**Comment by johnramsden** [comment]
Well, that helped, Now at least the error is gone but when I click the button to launch the KVM console nothing happens. 

`strace` shows the following when I launch the KVM console: 

```
[pid 16934] execve("/nix/store/fypifsmnf4yxqjs328dzd12h6j2v8sgq-IPMIView-20151223/jre/bin/java", ["java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "REDACTED", "null", "5900", "623", "0", "0"], [/* 90 vars */]) = 0
[pid 16934] +++ exited with 1 +++
[pid 16909] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=16934, si_uid=1000, si_status=1, si_utime=0, si_stime=0} ---
```

**Comment by Mic92** [comment]
I have no motherboard to debug this further. You could check the network traffic.

**Comment by johnramsden** [comment]
@Mic92 I was wondering if I might need to set the classpath as was done [here](https://github.com/NixOS/nixpkgs/blob/cf7a383b610e182af921188e5b67e45293d25c29/pkgs/development/tools/build-managers/gradle/default.nix#L17), but I wasn't sure what to set it to. Do you think that might be it? I don't know much about java, but it seems to be having trouble launching iKVM.jar.

**Comment by Mic92** [comment]
I suspect that the strace output of:

```
"java", "-Djava.library.path=.", "-jar", "iKVM.jar", "172.20.30.3", "ADMIN", "REDACTED", "null", "5900", "623", "0", "0"
```

is already correct.

**Comment by johnramsden** [comment]
OK, thanks for your help.

**Comment by johnramsden** [comment]
I guess if you're happy with it this could be committed, could it not? With the commit it at least works other than the KVM console.

**Comment by domenkozar** [comment]
For the record: I never got the KVM console to work inside the Nix package.

It has to do with library loading, something among `set LD_LIBRARY_PATH to $out`

**Comment by volth** [comment]
```patchelf``` of ```libiKVM64.so``` (a JNI library) has no effect:
```
$ ldd libiKVM64.so
 linux-vdso.so.1 (0x00007ffd45569000)
 libstdc++.so.6 => not found
 libm.so.6 => /nix/store/l48biijfr1j6d5kdg911051x2phfjrz7-glibc-2.25/lib/libm.so.6 (0x00007fd80a72d000)
 libgcc_s.so.1 => /nix/store/l48biijfr1j6d5kdg911051x2phfjrz7-glibc-2.25/lib/libgcc_s.so.1 (0x00007fd80a517000)
 libc.so.6 => /nix/store/l48biijfr1j6d5kdg911051x2phfjrz7-glibc-2.25/lib/libc.so.6 (0x00007fd80a178000)
 /nix/store/l48biijfr1j6d5kdg911051x2phfjrz7-glibc-2.25/lib64/ld-linux-x86-64.so.2 (0x00007fd80f369000)
```

There suppose to be something like ```--set-rpath "${gcc-unwrapped.lib}/lib64"``` instead of ```--set-rpath "${gcc.cc}/lib"```

**Comment by domenkozar** [comment]
Fixed in 41740ec0395059d426a4b3676c45cd886d2249ad thanks

**Pull request merged**